v1.2.5 build fails gcc 4.8

## NNG & Platform details.

NNG v1.2.5
Centos 5
devtoolset-2 (GCC 4.8): The C compiler identification is GNU 4.8.2
cmake version 3.12.20180626

## Expected Behavior

It would build with C99

## Actual Behavior

It does not build with C99.

I have looked at the other issues and there is comment which implies it would now build with older linux's (again, it built last in v1.2.2, in my testing).

## Steps to Reproduce

```
Unpack
cd nng-1.2.5
mkdir build && cd build
cmake .. # Don't have Ninja
make
```


```
[ 14%] Building C object src/CMakeFiles/nng.dir/platform/posix/posix_atomic.c.o
/nng/src-repo/src/platform/posix/posix_atomic.c:137:1: error: conflicting types for ‘nni_atomic_set_bool’
 nni_atomic_set_bool(nni_atomic_bool *b)
 ^
In file included from /nng/src-repo/src/core/nng_impl.h:27:0,
                 from /nng/src-repo/src/platform/posix/posix_atomic.c:13:
/nng/src-repo/src/core/platform.h:177:13: note: previous declaration of ‘nni_atomic_set_bool’ was here
 extern void nni_atomic_set_bool(nni_atomic_bool *, bool);
             ^
/nng/src-repo/src/platform/posix/posix_atomic.c: In function ‘nni_atomic_set_bool’:
/nng/src-repo/src/platform/posix/posix_atomic.c:140:3: error: dereferencing pointer to incomplete type
  b->b = false;
   ^
/nng/src-repo/src/platform/posix/posix_atomic.c: At top level:
/nng/src-repo/src/platform/posix/posix_atomic.c:145:1: error: conflicting types for ‘nni_atomic_get_bool’
 nni_atomic_get_bool(nni_atomic_bool *b)
 ^
In file included from /nng/src-repo/src/core/nng_impl.h:27:0,
                 from /nng/src-repo/src/platform/posix/posix_atomic.c:13:
/nng/src-repo/src/core/platform.h:178:13: note: previous declaration of ‘nni_atomic_get_bool’ was here
 extern bool nni_atomic_get_bool(nni_atomic_bool *);
             ^
/nng/src-repo/src/platform/posix/posix_atomic.c: In function ‘nni_atomic_get_bool’:
/nng/src-repo/src/platform/posix/posix_atomic.c:149:7: error: dereferencing pointer to incomplete type
  v = b->b;
       ^
/nng/src-repo/src/platform/posix/posix_atomic.c:151:2: warning: ‘return’ with a value, in function returning void [enabled by default]
  return (v);
  ^
/nng/src-repo/src/platform/posix/posix_atomic.c: At top level:
/nng/src-repo/src/platform/posix/posix_atomic.c:155:1: error: conflicting types for ‘nni_atomic_swap_bool’
 nni_atomic_swap_bool(nni_atomic_bool *b, bool n)
 ^
In file included from /nng/src-repo/src/core/nng_impl.h:27:0,
                 from /nng/src-repo/src/platform/posix/posix_atomic.c:13:
/nng/src-repo/src/core/platform.h:179:13: note: previous declaration of ‘nni_atomic_swap_bool’ was here
 extern bool nni_atomic_swap_bool(nni_atomic_bool *, bool);
             ^
/nng/src-repo/src/platform/posix/posix_atomic.c: In function ‘nni_atomic_swap_bool’:
/nng/src-repo/src/platform/posix/posix_atomic.c:159:7: error: dereferencing pointer to incomplete type
  v = b->b;
       ^
/nng/src-repo/src/platform/posix/posix_atomic.c:160:3: error: dereferencing pointer to incomplete type
  b->b = n;
   ^
/nng/src-repo/src/platform/posix/posix_atomic.c:162:2: warning: ‘return’ with a value, in function returning void [enabled by default]
  return (v);
  ^
/nng/src-repo/src/platform/posix/posix_atomic.c: In function ‘nni_atomic_init_bool’:
/nng/src-repo/src/platform/posix/posix_atomic.c:168:3: error: dereferencing pointer to incomplete type
  b->b = false;
   ^
make[2]: *** [src/CMakeFiles/nng.dir/platform/posix/posix_atomic.c.o] Error 1
make[1]: *** [src/CMakeFiles/nng.dir/all] Error 2
make: *** [all] Error 2
[  0%] Building C object src/CMakeFiles/nng.dir/platform/posix/posix_atomic.c.o
/nng/src-repo/src/platform/posix/posix_atomic.c:137:1: error: conflicting types for ‘nni_atomic_set_bool’
 nni_atomic_set_bool(nni_atomic_bool *b)
 ^
In file included from /nng/src-repo/src/core/nng_impl.h:27:0,
                 from /nng/src-repo/src/platform/posix/posix_atomic.c:13:
/nng/src-repo/src/core/platform.h:177:13: note: previous declaration of ‘nni_atomic_set_bool’ was here
 extern void nni_atomic_set_bool(nni_atomic_bool *, bool);
             ^
/nng/src-repo/src/platform/posix/posix_atomic.c: In function ‘nni_atomic_set_bool’:
/nng/src-repo/src/platform/posix/posix_atomic.c:140:3: error: dereferencing pointer to incomplete type
  b->b = false;
   ^
/nng/src-repo/src/platform/posix/posix_atomic.c: At top level:
/nng/src-repo/src/platform/posix/posix_atomic.c:145:1: error: conflicting types for ‘nni_atomic_get_bool’
 nni_atomic_get_bool(nni_atomic_bool *b)
 ^
In file included from /nng/src-repo/src/core/nng_impl.h:27:0,
                 from /nng/src-repo/src/platform/posix/posix_atomic.c:13:
/nng/src-repo/src/core/platform.h:178:13: note: previous declaration of ‘nni_atomic_get_bool’ was here
 extern bool nni_atomic_get_bool(nni_atomic_bool *);
             ^
/nng/src-repo/src/platform/posix/posix_atomic.c: In function ‘nni_atomic_get_bool’:
/nng/src-repo/src/platform/posix/posix_atomic.c:149:7: error: dereferencing pointer to incomplete type
  v = b->b;
       ^
/nng/src-repo/src/platform/posix/posix_atomic.c:151:2: warning: ‘return’ with a value, in function returning void [enabled by default]
  return (v);
  ^
/nng/src-repo/src/platform/posix/posix_atomic.c: At top level:
/nng/src-repo/src/platform/posix/posix_atomic.c:155:1: error: conflicting types for ‘nni_atomic_swap_bool’
 nni_atomic_swap_bool(nni_atomic_bool *b, bool n)
 ^
In file included from /nng/src-repo/src/core/nng_impl.h:27:0,
                 from /nng/src-repo/src/platform/posix/posix_atomic.c:13:
/nng/src-repo/src/core/platform.h:179:13: note: previous declaration of ‘nni_atomic_swap_bool’ was here
 extern bool nni_atomic_swap_bool(nni_atomic_bool *, bool);
             ^
/nng/src-repo/src/platform/posix/posix_atomic.c: In function ‘nni_atomic_swap_bool’:
/nng/src-repo/src/platform/posix/posix_atomic.c:159:7: error: dereferencing pointer to incomplete type
  v = b->b;
       ^
/nng/src-repo/src/platform/posix/posix_atomic.c:160:3: error: dereferencing pointer to incomplete type
  b->b = n;
   ^
/nng/src-repo/src/platform/posix/posix_atomic.c:162:2: warning: ‘return’ with a value, in function returning void [enabled by default]
  return (v);
  ^
/nng/src-repo/src/platform/posix/posix_atomic.c: In function ‘nni_atomic_init_bool’:
/nng/src-repo/src/platform/posix/posix_atomic.c:168:3: error: dereferencing pointer to incomplete type
  b->b = false;
   ^
```

--- a/src/platform/posix/posix_atomic.c
+++ b/src/platform/posix/posix_atomic.c
@@ -1,5 +1,5 @@
 //
-// Copyright 2019 Staysail Systems, Inc. <info@staysail.tech>
+// Copyright 2020 Staysail Systems, Inc. <info@staysail.tech>
 // Copyright 2018 Capitar IT Group BV <info@capitar.com>
 //
 // This software is supplied under the terms of the MIT License, a
@@ -204,10 +204,10 @@ nni_atomic_flag_reset(nni_atomic_flag *f)
 }
 
 void
-nni_atomic_set_bool(nni_atomic_bool *b)
+nni_atomic_set_bool(nni_atomic_bool *b, bool n)
 {
 	pthread_mutex_lock(&plat_atomic_lock);
-	b->b = false;
+	b->b = n;
 	pthread_mutex_unlock(&plat_atomic_lock);
 }
 

--- a/src/platform/posix/posix_impl.h
+++ b/src/platform/posix/posix_impl.h
@@ -1,5 +1,5 @@
 //
-// Copyright 2019 Staysail Systems, Inc. <info@staysail.tech>
+// Copyright 2020 Staysail Systems, Inc. <info@staysail.tech>
 // Copyright 2018 Capitar IT Group BV <info@capitar.com>
 //
 // This software is supplied under the terms of the MIT License, a
@@ -100,7 +100,7 @@ struct nni_atomic_flag {
 	bool f;
 };
 
-struct nni_atomic_bol {
+struct nni_atomic_bool {
 	bool b;
 };
 

